### ISSUE TITLE
Constant "found too many Xcode projects" error. I only have one.

### ISSUE THREAD

**Issue by jshier** [orta/cocoapods-keys#80]
I'm trying to integrate cocoapods-keys into a Swift project but I'm not getting beyond the initial setup. I installed it the first time and it worked fine, prompted for the key during `pod update` but output configuration warnings at the end of the update, so I deleted the configurations and ran `pod update` again. Now it just errors out with the "found too many Xcode projects" error. Even after I enter the name of the project it just keeps repeating the error. Here's my podfile (yes, I'm current trying JSON serialization libraries for Swift, this is a testing app):
```
platform :ios, '8.3'

use_frameworks!

target 'WeatherSwift' do
  pod 'Alamofire'
  pod 'AlamofireObjectMapper'
  pod 'Argo'
  pod 'Mantle'
end

target 'WeatherSwiftTests' do
  pod 'Nimble'
  pod 'OHHTTPStubs'
  pod 'Quick'
end

plugin 'cocoapods-keys', {
  :keys => [
    "ForecastAPIKey"
  ]
}
```

**Comment by jshier** [comment]
Well, I got the error to go away by adding the project name directly to the Podfile. Isn't the detection supposed to be automatic? However, now whenever I try to do a `pod update` after adding keys to the Podfile, I get the following warning output after `pod update`:
```
[!] CocoaPods did not set the base configuration of your project because your project already has a custom config set. In order for CocoaPods integration to work at all, please either set the base configurations of the target `WeatherSwift` to `Pods/Target Support Files/Pods/Pods.debug.xcconfig` or include the `Pods/Target Support Files/Pods/Pods.debug.xcconfig` in your build configuration.

[!] CocoaPods did not set the base configuration of your project because your project already has a custom config set. In order for CocoaPods integration to work at all, please either set the base configurations of the target `WeatherSwift` to `Pods/Target Support Files/Pods/Pods.release.xcconfig` or include the `Pods/Target Support Files/Pods/Pods.release.xcconfig` in your build configuration.
```
Pods.*.xcconfig files don't exist and previously it was asking for the WeatherSwift-Pods.*.xcconfig files. I try my standard fix for this error and delete the configurations from the project's interface and then I get these warnings:
```
[!] CocoaPods did not set the base configuration of your project because your project already has a custom config set. In order for CocoaPods integration to work at all, please either set the base configurations of the target `WeatherSwift` to `Pods/Target Support Files/Pods-WeatherSwift/Pods-WeatherSwift.debug.xcconfig` or include the `Pods/Target Support Files/Pods-WeatherSwift/Pods-WeatherSwift.debug.xcconfig` in your build configuration.

[!] CocoaPods did not set the base configuration of your project because your project already has a custom config set. In order for CocoaPods integration to work at all, please either set the base configurations of the target `WeatherSwift` to `Pods/Target Support Files/Pods-WeatherSwift/Pods-WeatherSwift.release.xcconfig` or include the `Pods/Target Support Files/Pods-WeatherSwift/Pods-WeatherSwift.release.xcconfig` in your build configuration.
```
And they never go away, no matter how much I delete the configurations from the project. Seems like this plugin shouldn't be affecting this stuff, right?

**Comment by orta** [comment]
Yeah, in this case it is creating that Pods target you need to set the target in your keys.

```
plugin 'cocoapods-keys', {
  :keys => [
    "ForecastAPIKey"
  ],
  :target => "WeatherSwift"
}
```

**Comment by orta** [comment]
I wonder how we can protect / warn against this

**Comment by jshier** [comment]
Thanks, that got it working, once I was able to remove the space that somehow got introduced at the beginning of my key.

**Comment by orta** [comment]
I improved error handling around this too, in addition to our better project handling. Should be fixed as of 1.5.0.
